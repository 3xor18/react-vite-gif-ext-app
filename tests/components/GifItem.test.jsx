import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Test <GifItem />', () => {

    const title = 'Saitama';
    const url = 'https://one.punchman.com/saitama.jpg'


    test('coincide con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        //con la u se actualiza
        expect(container).toMatchSnapshot()
    })

    test('Debe mostrar la imagen y laurl con el alt indicado ', () => { 
        render(<GifItem title={title} url={url} />)
        //screen.debug()
        const {src,alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
     })

     test('Debe de Mostrar el titulo del componente', () => { 
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
      })


})