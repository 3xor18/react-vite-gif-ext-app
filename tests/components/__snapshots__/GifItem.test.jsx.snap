// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifItem /> coincide con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one.punchman.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
